CHANGELOG.md

```diff
@@ -72,15 +99,96 @@ All notable changes to this project will be documented in this file.
+* i686-linux-android ✅ (⚠️)
+* x86_64-linux-android ✅ (⚠️)
+* aarch64-apple-ios ✅ (⚠️)
+* aarch64-apple-ios-sim ✅ (⚠️)
+* x86_64-apple-ios ✅ (⚠️)
+* wasm32-unknown-unknown ✅
+
+Note: all the platforms marked with (⚠️) above will get a round of polish for the build process & documentation in the next release.
+
+### CoreCrypto
+
+* Majorly improved documentation across all crates. Documentation for the `main` branch can be found here. The `HEAD` of this branch should only be a tagged version.
+    * This documentation is available here: <https://wireapp.github.io/core-crypto/core_crypto/>
+* Moved the codebase to `async`
+    * This was a requirement to make everything work on the WASM target, as we cannot block the JavaScript runtime without making the browsers freeze up completely
+    * As a consequence, we forked `openmls` to [wireapp/openmls](https://github.com/wireapp/openmls)
+        * Our incremental changes, including the `async` rewrite of `openmls` is located [here](https://github.com/wireapp/openmls/pull/4)
+* Added support for MLS Group Persistence, as this was preventing clients from continuing use of their joined groups (oops!)
+* Added support for lifetime-expired Keypackage pruning
+* Added support for external CSPRNG entropy pool seeding
+* Dropped the `openmls-rust-crypto-provider` in favour of our `mls-crypto-provider` with support for more ciphersuites and updated dependencies
+    * As a consequence, we forked `hpke-rs` to [wireapp/hpke-rs](https://github.com/wireapp/hpke-rs)
+        * Our changes can be found [here](https://github.com/wireapp/hpke-rs/tree/fix/updated-deps)
+    * Ciphersuite support details:
+        * `MLS_128_DHKEMX25519_AES128GCM_SHA256_Ed25519` ✅
+        * `MLS_128_DHKEMP256_AES128GCM_SHA256_P256` ✅
+        * `MLS_128_DHKEMX25519_CHACHA20POLY1305_SHA256_Ed25519` ✅
+        * `MLS_256_DHKEMX448_AES256GCM_SHA512_Ed448` ❌
+            * There is no suitable `ed448` rust crate yet
+        * `MLS_256_DHKEMP521_AES256GCM_SHA512_P521` ❌
+            * `p521` RustCrypto crate is a WIP and not ready just yet. It shouldn't take too long though.
+        * `MLS_256_DHKEMX448_CHACHA20POLY1305_SHA512_Ed448` ❌
+            * There is no suitable `ed448` rust crate yet
+        * `MLS_256_DHKEMP384_AES256GCM_SHA384_P384` ✅
+
+* Expanded the API to include:
+    * Conversations:
+        * Ability to wipe
+        * Ability to leave
+        * Ability to force clients to update their keying material (i.e. self-update)
+    * Support for MLS proposals
+    * Support for MLS external commits
+        * Added ability to export MLS Public Group State for a given conversation
+        * Added support for creating an external commit to join a conversation
+    * Support for MLS external Add and Remove Proposal support
+    * Support for X.509 credentials
+    * Added a commit delay hint to prevent clients from rushing to commit to the server - which would cause epoch conflicts and high load
```

Comment:
```suggestion
    * Added a commit delay hint to prevent clients from rushing to commit to the server - which would cause epoch conflicts and high load. Returned in `decrypt_message`.
```